toolkit/resources/assets/efi/grub/grub.cfg

```diff
@@ -3,4 +3,8 @@ search -n -u {{.BootUUID}} -s
 # If '/boot' is a seperate partition, BootUUID will point directly to '/boot'.
 # In this case we should omit the '/boot' prefix from all paths.
 set bootprefix={{.BootPrefix}}
+# For images using grub2-mkconfig, $prefix is the variable
+# We need to populate with the proper path to menuentry grub.cfg,
+# grubenv
+set prefix=($root)'/boot/grub2'
```

Comment:
can we stop manually setting prefix? grub should detect it correctly and us manually setting it only adds brittleness to our distro for no reason...

i'm not sure what 'bootprefix' does either but i suspect we shouldn't be setting that either...

Reply:
As far as I can tell, the prefix grub computes is a function of the where the grub efi binary is installed. It is effectively hard-coded in grub itself. 

Now my reading of this code could be totally wrong, but afaik the prefix path is set here, in main. Said path comes from the fwpath variable, which is found here:

https://github.com/rhboot/grub2/blob/grub-2.06/grub-core/kern/main.c#L215

This looks like it is essentially finding the path of the EFI binary which was loaded. 

https://github.com/rhboot/grub2/blob/grub-2.06/grub-core/kern/main.c#L129
https://github.com/rhboot/grub2/blob/fedora-39/grub-core/kern/efi/init.c#L172

That would be grubx64.efi, found roughly here in azl: 
![image](https://github.com/microsoft/CBL-Mariner/assets/37827103/beb28acd-322e-43ad-9eaa-6c699e8ae63e)

If all of that holds true, our prefix variable is still completely wrong until we set it otherwise.